### PR TITLE
Add new reader for GCS

### PR DESCRIPTION
**Pull request by nfiacco**
# Description

New reader for GCS, uses same structure as S3. Using [gcsfs](https://gcsfs.readthedocs.io/en/latest/) to provide FS interface for the SimpleDirectoryReader.

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [x] Yes
- [ ] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [x] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Comment by nfiacco** [comment]
May need help creating a test notebook @Disiok et. al

File: llama-index-integrations/readers/llama-index-readers-gcs/poetry.lock
Comment (logan-markewich) [inline llama-index-integrations/readers/llama-index-readers-gcs/poetry.lock:1]: No need to commit the lock file 👍🏻

File: llama-index-integrations/readers/llama-index-readers-gcs/llama_index/readers/gcs/base.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        """
+        Load the file(s) from GCS.
+
+        Args:
+            custom_temp_subdir (str, optional): This parameter is deprecated and unused. Defaults to None.
+
+        Returns:
+            List[Document]: A list of documents loaded from GCS.
+        """
+        if custom_temp_subdir is not None:
```
Comment (logan-markewich) [inline llama-index-integrations/readers/llama-index-readers-gcs/llama_index/readers/gcs/base.py:116]: If this is a new package, we don't need to include unused parameters?

Comment (nfiacco) [reply llama-index-integrations/readers/llama-index-readers-gcs/llama_index/readers/gcs/base.py:116]: 🤦

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**